src/Json_encode.ml

```diff
@@ -17,4 +21,7 @@ let list encode l =
 
 external stringArray : string array -> Js.Json.t = "%identity"
 external numberArray : float array -> Js.Json.t = "%identity"
-external booleanArray : Js.boolean array -> Js.Json.t = "%identity"
\ No newline at end of file
+
+external jsbooleanArray : Js.boolean array -> Js.Json.t = "%identity"
+
+let booleanArray arr = arr |> Array.map Js.Boolean.to_js_boolean |> jsbooleanArray
```

Comment:
The only reason `booleanArray` (and the other `*Array` functions) exist is to avoid the performance hit of needlessly iterating through an array to cast each element. Since we already know the type is valid Json, we can just cast it directly.

This should instead be accomplished by something like `composableArray bool`, I'm just not yet sure what a good name for such a function would be. Or perhaps the existing `array` should be replaced and renamed to something like `arrayOfJson`.

Reply:
OK I just changed it for consistency.

Reply:
I figured. It's probably a good time to figure out how to do this properly anyway though. I'll think about it a bit, but please chime in if you have any opinion on the naming or approach in general.